src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1118,19 +1128,130 @@ map_injective f.injective u v
 
 end path
 
+/-! ## Transferring between graphs -/
+
+namespace walk
+
+variables {G}
+
+/-- The walk `p` transferred to lie in `H` -/
+@[protected, simp] def transfer : Π {u v : V} (p : G.walk u v) (H : simple_graph V)
+  (h : ∀ e, e ∈ p.edges → e ∈ H.edge_set), H.walk u v
+| _ _ (walk.nil) H h := walk.nil
+| _ _ (walk.cons a p) H h := by
+  { refine walk.cons _ (p.transfer H _);
+    simp only [walk.edges_cons, list.mem_cons_iff, forall_eq_or_imp, mem_edge_set] at h,
+    exact h.1, exact h.2, }
+
+variables {u v w : V} (p : G.walk u v) (q : G.walk v w)
+  {H : simple_graph V}
+  (hp : ∀ e, e ∈ p.edges → e ∈ H.edge_set)
+  (hq : ∀ e, e ∈ q.edges → e ∈ H.edge_set)
+
+lemma transfer_id : p.transfer G (λ e ep, edges_subset_edge_set p ep) = p := by
+begin
+  induction p,
+  { simp only [transfer] },
+  { simp only [p_ih, transfer, eq_self_iff_true, heq_iff_eq, and_self] },
+end
```

Comment:
Thanks for putting these into mathlib style. Now that you've done it, let me ask you to possibly change them again :wink:

I noticed that this and many other proofs can be replaced by
```suggestion
by induction p; simp [*]
```
Sometimes you might need additional simp lemmas, but you can switch that simp to `simp [*, extra_lemma]`.

Reply:
The previous style was actually a remnant of one-line `simp` heavy proofs as you're using, but I thought I'd rather have `simp only`. I'll revert the changes

Reply:
You can still do a `simp only` if you want. You can do `squeeze_scope { induction p; squeeze_simp [*] }` to get the simp lemmas you need, but you might need to manually dedup the simp list.

Reply:
One way to think of these `induction p; simp [*]` proofs is that they strongly indicate "the proof is trivial." For these types of proofs, I usually only squeeze my simps if either (1) there's a small list of simp lemmas and knowing them is informative or (2) the simp takes longer than a second, since mathlib compilation time is important.